docs/src/pages/components/data-grid/editing/editing.md

```diff
@@ -5,7 +5,129 @@ components: DataGrid, XGrid
+The editable cells have a green background for better visibility.
+
+{{"demo": "pages/components/data-grid/editing/IsCellEditableGrid.js", "bg": "inline"}}
+
+### Control editing
+
+The `editRowsModel` prop lets you control the editing state.
+You can handle the `onEditRowModelChange` callback to control the `GridEditRowsModel` state.
+
+{{"demo": "pages/components/data-grid/editing/EditRowModelControlGrid.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Column with valueGetter
+
+You can control the committed value when the edit move stops with the `onEditCellChangeCommitted` prop.
+This is especially interesting when using the `valueGetter` on the column definition.
+
+{{"demo": "pages/components/data-grid/editing/ValueGetterGrid.js", "bg": "inline"}}
+
+### Client-side validation
+
+Follow the following steps to validate the value in the cells.
+
+- Set the event handler `onEditCellChange`. It's invoked when a change is triggered by the edit input component.
+- Set the event handler `onEditCellChangeCommitted` to validate or persist the new value. This handler is invoked when an end-user requests a change to be committed and before the cell reverts to view mode.
+
+Alternatively, you can use the `GridEditRowsModel` state mentioned in the [Control editing](#control-editing) section.
+
+{{"demo": "pages/components/data-grid/editing/ValidateRowModelControlGrid.js", "bg": "inline"}}
+
+#### Using apiRef [<span class="pro"></span>](https://material-ui.com/store/items/material-ui-pro/)
+
+{{"demo": "pages/components/data-grid/editing/ValidateCellApiRefGrid.js", "bg": "inline"}}
+
+### Server-side validation
+
+Server-side validation works like client-side [validation](#validation).
+
+- Use the `GridEditRowsModel` mentioned in the [Control editing](#control-editing) section.
+- Or set the event handler `onEditCellChange` for `keydown` validation
+- Set the event handler `onEditCellChangeCommitted` to validate and update the server when the change is committed.
+
+**Note:** To prevent the default client-side behavior, use `event.stopPropagation()`.
+
+This demo shows how you can validate a username asynchronously and prevent the user from committing the value while validating.
+
+{{"demo": "pages/components/data-grid/editing/ValidateServerNameGrid.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Custom edit component
+
+To customize the edit component of a column, set the `renderEditCell` function available in the column definition `GridColDef`.
+
+The demo lets you edit the ratings by double-clicking the cell.
+
+{{"demo": "pages/components/data-grid/editing/RenderRatingEditCellGrid.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Edit using external button
+
+You can override the default [start editing](#start-editing) triggers using the `event.stopPropagation()` API on the synthetic React events.
+
+{{"demo": "pages/components/data-grid/editing/StartEditButtonGrid.js", "bg": "inline"}}
```

Comment:
A side note not directly related to this PR. Enforcing the focus on the cell seems plain wrong:

![focus](https://user-images.githubusercontent.com/3165635/114694101-815d0680-9d1a-11eb-8a89-3f0441b20f3f.gif)